app/src/main/java/org/greenstand/android/TreeTracker/fragments/TreeHeightFragment.kt

```diff
@@ -59,22 +62,35 @@ class TreeHeightFragment : Fragment() {
             vm.saveNewTree()
         }
 
-        vm.toastMessagesLiveData().observe(this, Observer {
-            Toast.makeText(context, it, Toast.LENGTH_SHORT).show()
-        })
+        vm.toastMessagesLiveData().observe(
+            this,
+            Observer {
+                Toast.makeText(context, it, Toast.LENGTH_SHORT).show()
+            }
+        )
+
+        vm.onFinishedLiveData().observe(
+            this,
+            Observer {
+                findNavController().popBackStack(R.id.mapsFragment, false)
+            }
+        )
 
-        vm.onFinishedLiveData().observe(this, Observer {
-            findNavController().popBackStack(R.id.mapsFragment, false)
-            fragmentManager?.popBackStack(ValueHelper.NEW_TREE_FRAGMENT, FragmentManager.POP_BACK_STACK_INCLUSIVE)
-        })
+        vm.onEnableButtonLiveData().observe(
+            this,
+            Observer {
+                save_tree_height.isEnabled = it
+            }
+        )
 
-        vm.onEnableButtonLiveData().observe(this, Observer {
-            save_tree_height.isEnabled = it
-        })
+        requireActivity().onBackPressedDispatcher.addCallback(this) {
```

Comment:
This is change that traps the event when back button is pressed and takes the user to the maps screen.